raster_loader/io/common.py

```diff
@@ -317,141 +363,309 @@ def band_with_nodata_mask(
+            [band_and_mask[1] for band_and_mask in bands_and_masks]
+        )
+        raw_data = bands_and_masks[compound_bands.index(band)][0]
     else:
-        masked = True
-        if alpha_band:
-            # mask data with alpha band to exclude from stats
-            stats = read_masked(raster_dataset, band, alpha_band)
-        else:
-            # mask nodata values to exclude from stats
-            compound_bands = get_compound_bands(raster_dataset, band)
-            if len(compound_bands) > 1:
-                # if band is part of a RGB triplet,
-                # we need to use the three bands for masking
-                bands_and_masks = [
-                    band_with_nodata_mask(raster_dataset, b) for b in compound_bands
-                ]
-                mask = np.logical_and.reduce(
-                    [band_and_mask[1] for band_and_mask in bands_and_masks]
-                )
-                raw_data = bands_and_masks[compound_bands.index(band)][0]
-            else:
-                (raw_data, mask) = band_with_nodata_mask(raster_dataset, band)
-            stats = np.ma.masked_array(data=raw_data, mask=mask)
-    qdata = stats.compressed()
-    ranges = [[j / i for j in range(1, i)] for i in range(3, 20)]
-    casting_function = int if np.issubdtype(stats.dtype, np.integer) else float
-    quantiles = [
-        [casting_function(np.quantile(qdata, q, method="lower")) for q in r]
-        for r in ranges
-    ]
-    quantiles = dict(zip(range(3, 20), quantiles))
-    most_common = Counter(qdata).most_common(100)
-    most_common.sort(key=lambda x: x[1], reverse=True)
-    most_common = dict([(casting_function(x[0]), x[1]) for x in most_common])
+        (raw_data, mask) = band_with_nodata_mask(raster_dataset, band)
+
+    return np.ma.masked_array(data=raw_data, mask=mask)
+
+
+def raster_band_stats(
+    raster_dataset: rasterio.io.DatasetReader, band: int, all_stats: bool
+) -> dict:
+    """Get statistics for a raster band."""
+
+    print('Computing stats for band {0}...'.format(band))
+    _stats = raster_dataset.stats(indexes=[band], approx=False)[0]
+    _min = _stats.min
+    _max = _stats.max
+    _mean = _stats.mean
+    _std = _stats.std
+
+    count = math.prod(_stats.shape)
+    if is_masked_band(raster_dataset, band):
+        count = np.count_nonzero(_stats.mask is False)
+
+    _sum = _mean * count
+    sum_squares = count * _std ** 2 + _mean ** 2
+
+    quantiles = None
+    most_common = None
+    if all_stats:
+        raster_band = read_raster_band(raster_dataset=raster_dataset, band=band)
+
+        print("Removing masked data...")
+        qdata = raster_band.compressed()
+
+        casting_function = (
+            int if np.issubdtype(raster_band.dtype, np.integer) else float
+        )
+
+        quantiles = compute_quantiles(qdata, casting_function)
+
+        print("Computing most commons values...")
+        # Not sure whether we should compute most_common values for float bands
```

Comment:
Maybe a warning could be shown to the user in this case, so they are aware of potentially meaningless/useless results

Reply:
I added the warning message as suggested!